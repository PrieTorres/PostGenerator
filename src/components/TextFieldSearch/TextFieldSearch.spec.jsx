import { render, screen } from "@testing-library/react"
import { TextFieldSearch } from "."
import '@testing-library/jest-dom';
import '@testing-library/jest-dom/extend-expect';
import userEvent from "@testing-library/user-event";

describe("<TextFieldSearch />", () => {
  it("should render with props", () => {
    const fn = jest.fn();
    render(<TextFieldSearch onChange={fn} value={"test"} />);
    const input = screen.getByPlaceholderText(/burcar.../i)

    expect(input).toBeInTheDocument();
    expect(input.value).toBe("test");
  });

  it("should call handleChangeFunction on each key pressed", () => {
    const fn = jest.fn();
    render(<TextFieldSearch onChange={fn} />);
    const input = screen.getByPlaceholderText(/burcar.../i);
    const value = "Lorem ipsum";

    userEvent.type(input, value);

    expect(input.value).toBe(value);
    expect(fn).toHaveBeenCalledTimes(value.length);
  });

  it("should change value when backspacing text", () => {
    const fn = jest.fn();
    render(<TextFieldSearch onChange={fn} />);
    const input = screen.getByPlaceholderText(/burcar.../i);
    const value = "Lorem ipsum";

    userEvent.type(input, value);

    expect(input.value).toBe(value);
    expect(fn).toHaveBeenCalledTimes(value.length);

    userEvent.type(input, "{backspace}");

    expect(input.value).toBe("Lorem ipsu");
    expect(fn).toHaveBeenCalledTimes(value.length+1);
  });

  it("should match snapshot", () => {
    const fn = jest.fn();
    const {container} = render(<TextFieldSearch onChange={fn} value={"test"} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
