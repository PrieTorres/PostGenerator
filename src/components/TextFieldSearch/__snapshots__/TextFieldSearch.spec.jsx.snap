// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextFieldSearch /> should match snapshot 1`] = `
<div
  class="container"
>
  <input
    class="input"
    placeholder="burcar..."
    type="search"
    value="test"
  />
</div>
`;
